scripts/west_commands/runners/core.py

```diff
@@ -483,12 +485,15 @@ class ZephyrBinaryRunner(abc.ABC):
     commands in its constructor.  The actual command execution is
     handled in the run() method.'''
 
-    def __init__(self, cfg: RunnerConfig):
+    def __init__(self, cfg: RunnerConfig, dry_run: bool=_DRY_RUN):
```

Comment:
@pdgendt I am fairly sure this will not break out-of-tree runners, but FYI. Also, CC @mbolivar.

Reply:
@mbolivar since we are very close to release I've opted to make this specific to a single runner. I may try to generalize this later if it's worth it, because I had to duplicate a tiny bit of code.

Reply:
Nothing at all, all good. I just wanted to make sure to get this in for 4.3.0 @mbolivar. Will follow-up next week.